### ISSUE TITLE
Using `RootInspector<D>` instead of `D` as parameter in `validate()` function 

### ISSUE THREAD

**Issue by jamowei** [jwstegemann/fritz2#368]
Instead of using the model of type `D` directly we could wrap it always to a `RootInspector<D>` by using the `inspect()` function.
```kotlin
expect abstract class Validator<D, M : ValidationMessage, T>() {
    abstract fun validate(data: RootInspector<D>, metadata: T): List<M>
}
```
Normally you call then the `isValid()` function with your current data model. This function is implemented for different target but only needs to call the `inspect()` function before.
```kotlin
//jvm side
fun isValid(data: D, metadata: T): Boolean = validate(inspect(data), metadata)
        .also { msgs = it }.none(ValidationMessage::isError)

//js side
fun isValid(data: D, metadata: T): Boolean {
    val messages = validate(inspect(data), metadata)
    state.value = messages
    return messages.none(ValidationMessage::isError)
}
```
Creating your own `Validator` then looks like:
```kotlin
val carValidator = object: Validator<Car, Message, Unit>() {
    override fun validate(inspector: RootInspector<Car>, metadata: Unit): List<Message> {
        val msgs = mutableListOf<Message>()

        if (inspector.data.name.isBlank())
            msgs.add(Message("car name can not be blank"))

        if (inspector.data.color.r < 0 || inspector.data.color.g < 0 || inspector.data.color.b < 0)
            msgs.add(Message("color members are lower then 0"))

        if (inspector.data.color.r > 255 || inspector.data.color.g > 255 || inspector.data.color.b > 255)
            msgs.add(Message("color members are greater then 255"))

        return msgs
    }
}
```

When using the `ComponentValidator` and `ComponentValidationMessage` you can't forget to create the `RootInspector` first and your `id` which are important for matching the right element are always correct.

**Comment by jamowei** [comment]
We can also use a `MutableList` context instead (same as `buildList()` function). Then there is no need to create your own list inside the `validate()` function anymore.

**Comment by jamowei** [comment]
Solved by #570
